api/hypershift/v1beta1/gcp.go

```diff
@@ -0,0 +1,31 @@
+package v1beta1
+
+// GCPPlatformSpec specifies configuration for clusters running on Google Cloud Platform.
+type GCPPlatformSpec struct {
+	// project is the GCP project ID.
+	// A valid project ID must satisfy the following rules:
+	//   length: Must be between 6 and 30 characters, inclusive
+	//   characters: Only lowercase letters (`a-z`), digits (`0-9`), and hyphens (`-`) are allowed
+	//   start and end: Must begin with a lowercase letter and must not end with a hyphen
+	//   valid examples: "my-project", "my-project-1", "my-project-123".
+	//
+	// +required
+	// +immutable
```

Comment:
this marker was a convention. If you want this field to be immutable if should be enforced via cel, which I don't see here